src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,144 @@
+    private static final String SQL_FIND_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SQL_FIND_ALL = "SELECT * FROM books;";
+    private static final String SQL_UPDATE = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String SQL_DELETE = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_CREATE,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            verifyAffectedRows(statement.executeUpdate());
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add book " + book.getTitle() + "to the DB", e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_FIND_BY_ID)) {
+            statement.setLong(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+
+            book = parseBook(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find a book with id " + id, e);
+        }
+
+        return Optional.ofNullable(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_FIND_ALL)) {
+            ResultSet allBooks = statement.executeQuery();
+
+            return getBooks(allBooks, books);
+        } catch (SQLException e) {
+            throw new DataProcessingException("DB is empty", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        Long id = book.getId();
+        String title = book.getTitle();
+        BigDecimal price = book.getPrice();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_UPDATE)) {
+            statement.setString(1, title);
+            statement.setBigDecimal(2, price);
+            statement.setLong(3, id);
+
+            statement.executeUpdate();
+
+            Optional<Book> oldBook = findById(id);
+
+            return oldBook.orElseThrow(()
+                    -> new RuntimeException("Book is not exist"));
```

Comment:
I guess we do not need this part of the code. We can simply return `book` if everything is successful. Otherwise, `SQLException` will be thrown and we've already caught that one